stdlib/set.mli

```diff
@@ -118,6 +118,17 @@ module type S =
        The elements of [s] are presented to [f] in increasing order
        with respect to the ordering over the type of the elements. *)
 
+    val map: (elt -> elt) -> t -> t
+    (** [map f s] creates a new set with elements [f a0],
+        [f a1]... [f aN], where [a0],[a1]..[aN] are the
+        values contained in [s].
+
+       The elements are passed to [f] in increasing order
+       with respect to the ordering over the type of the keys.
+
+       If [f] is the identity over the elements of [s], the returned
+       set is physically equal to [s]. *)
```

Comment:
This is done for the other functions for which the guarantee is provided. In the present case, it does constrain the possible implementation approaches (adding in a fold does not respect this requirement), which would be an argument against. But if I took the care of providing this guarantee in the implementation, I may as well let users rely on it.

Note that this is weaker that one may hope, for example it says nothing of sets were "almost all" elements are mapped to themselves, but some are changed.